tests/unit/test_req.py

```diff
@@ -460,3 +469,132 @@ def test_filter_install():
         INFO, 'foo bar')
     assert _filter_install('I made a SyntaxError') == (
         INFO, 'I made a SyntaxError')
+
+
+def test_join_line_continuations():
+    """
+    Test joining of line continuations
+    """
+
+    lines = dedent('''\
+    line 1
+    line 2:1 \\
+    line 2:2
+    line 3:1 \\
+    line 3:2 \\
+    line 3:3
+    line 4
+    ''').splitlines()
+
+    expect = [
+        'line 1',
+        'line 2:1 line 2:2',
+        'line 3:1 line 3:2 line 3:3',
+        'line 4',
+    ]
+
+    assert expect == list(join_lines(lines))
+
+
+def test_parse_editable_from_requirements():
+    lines = [
+        '--editable svn+https://foo#egg=foo',
+        '--editable=svn+https://foo#egg=foo',
+        '-e svn+https://foo#egg=foo'
+    ]
+
+    res = [parse_line(i) for i in lines]
+    assert res == [(REQUIREMENT_EDITABLE, 'svn+https://foo#egg=foo')] * 3
+
+    req = next(parse_content('fn', lines[0]))
+    assert req.name == 'foo'
+    assert req.link.url == 'svn+https://foo#egg=foo'
+
+
+@pytest.fixture
+def session():
+    return PipSession()
+
+
+@pytest.fixture
+def finder(session):
+    return PackageFinder([], [], session=session)
+
+
+def test_parse_options_from_requirements(finder):
+    assert parse_line('-i abc') == (OPTION, ('-i', 'abc'))
+    assert parse_line('-i=abc') == (OPTION, ('-i', 'abc'))
+    assert parse_line('-i  =  abc') == (OPTION, ('-i', 'abc'))
+
+    assert parse_line('--index-url abc') == (OPTION, ('--index-url', 'abc'))
+    assert parse_line('--index-url=abc') == (OPTION, ('--index-url', 'abc'))
+    assert parse_line('--index-url   =   abc') == (
+        OPTION, ('--index-url', 'abc')
+    )
+
+    with pytest.raises(RequirementsFileParseError):
+        parse_line('--allow-external')
+
+    res = parse_line('--extra-index-url 123')
+    assert res == (OPTION, ('--extra-index-url', '123'))
+
+    next(parse_content('fn', '-i abc', finder=finder), None)
+    assert finder.index_urls == ['abc']
+
+
+def test_parse_flags_from_requirements(finder):
+    assert parse_line('--no-index') == (FLAG, ('--no-index'))
+    assert parse_line('--no-use-wheel') == (FLAG, ('--no-use-wheel'))
+
+    with pytest.raises(RequirementsFileParseError):
+        parse_line('--no-use-wheel true')
+
+    next(parse_content('fn', '--no-index', finder=finder), None)
+    assert finder.index_urls == []
+
+
+def test_get_requirement_options():
+    res = parse_requirement_options('--install-option="--abc --zxc"')
+    assert res == {'install_options': ['--abc --zxc']}
+
+    res = parse_requirement_options('--global-option "--abc"')
+    assert res == {'global_options': ['--abc']}
+
+    line = (
+        'INITools==2.0 '
+        '--global-option="--one --two -3" '
```

Comment:
Sorry @pfmoore , I mean if pip should attempt to catch `--global-option='--author --author-email'` as an error before handing it off to `setup.py`. Your previous comment led me to believe so:

> Agreed - supplying --global-option='--author --author-email' should be an error rather than passing 2 arguments. At least that's how I believe the command line --global-option works (the docs aren't that clear) and it's what I think makes sense.

I agree on the `as is` passing of arguments and the tests verify exactly that behavior (i.e. `"--abc --zxc"` is parsed as a single argument). However, I'll change them on premise that they may indeed be misleading.